starter-code/src/functions-and-arrays.js

```diff
@@ -65,6 +87,27 @@ const averageWordLength = words => {
 };
 averageWordLength(wordsArr);
 
+// bonus 4.1
+const avg = arr => {
+  let counter = 0;
+  if (arr.length === 0) {
+    return null;
+  } else {
+    for (i = 0; i < arr.length; i++) {
+      if (typeof arr[i] === "number") {
+        counter += arr[i];
+      } else if (typeof arr[i] === "boolean") {
+        counter += arr[i] ? 1 : 0;
+      } else if (typeof arr[i] === "string") {
+        counter += arr[i].length;
+      }
+    }
+    return counter / arr.length;
+  }
+};
+
+console.log(avg(mixedArr));
+
 // Iteration #5: Unique arrays
```

Comment:
el código quedaría así para la solución de la iteración 5
`uniquifyArray=(arr)=>{

  if (arr.length !== 0) {
    let uniqueArr = [];

    for (word in arr) {
      if (
        arr.indexOf(arr[word]) !== -1 &&
        !uniqueArr.includes(arr[word])
      ) {
        uniqueArr.push(arr[word]);
      }
    }
    return uniqueArr;
  } else {
    return null;
  }`